servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CelAccessChecker.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.authz;
+
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.ASSIGN_REFERENCE_TO_HASH;
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.COMMIT_CHANGE_AGAINST_REFERENCE;
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.CREATE_REFERENCE;
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.DELETE_ENTITY;
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.DELETE_REFERENCE;
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.LIST_COMMIT_LOG;
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.READ_ENTITY_VALUE;
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.READ_ENTRIES;
+import static org.projectnessie.server.authz.CelAccessChecker.AuthorizationRuleType.UPDATE_ENTITY;
+
+import com.google.common.collect.ImmutableMap;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.ForbiddenException;
+import org.projectnessie.cel.tools.ScriptException;
+import org.projectnessie.model.ContentsKey;
+import org.projectnessie.server.config.QuarkusNessieAuthorizationConfig;
+import org.projectnessie.services.authz.AccessChecker;
+import org.projectnessie.services.authz.AccessContext;
+import org.projectnessie.services.cel.CELUtil;
+import org.projectnessie.versioned.NamedRef;
+
+/**
+ * A reference implementation of the {@link AccessChecker} that performs access checks using CEL
+ * expressions.
+ */
+@ApplicationScoped
+public class CelAccessChecker implements AccessChecker {
+
+  private final QuarkusNessieAuthorizationConfig config;
+
+  public enum AuthorizationRuleType {
```

Comment:
The static import above of this enum looks strange to my eye. Can it be simplified?

Reply:
done